File: src/main/java/com/example/blog/Service/PostService.java
package com.example.blog.Service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.blog.Entity.Post;
import com.example.blog.Repository.PostRepository;

import java.util.List;

@Service
public class PostService {

    @Autowired
    private PostRepository postRepository;

    public Post createPost(Post post) {
        if (post != null) {
            return postRepository.save(post);
        }
        return null;
    }

    public List<Post> getAllPosts() {
        return postRepository.findAll();
    }

    public List<Post> getPostsByCategory(String category) {
        return postRepository.findByCategory(category);
    }

    public void deletePost(Long id) {
        if (id != null) {
            postRepository.deleteById(id);
        }
    }
}

File: blogapp_Frontend/src/Services/postService.js
import api from "./api";

export const getAllPosts = async () => {
  const res = await api.get("/posts/all");
  return res.data;
};

export const getPostsByCategory = async (category) => {
  const res = await api.get(`/posts/category/${category}`);
  return res.data;
};

export const createPost = async (postData) => {
  const res = await api.post("/posts/create", postData);
  return res.data;
};

export const deletePost = async (id) => {
  const res = await api.delete(`/posts/delete/${id}`);
  return res.data;
};